react/error.tsx

```diff
@@ -17,25 +17,51 @@ class ErrorPage extends Component {
   public state = { enabled: false }
 
   public componentDidMount() {
-    window.setTimeout(() => {
-      this.setState({ enabled: true })
-    }, 5000)
+    try {
+      const error = window?.__ERROR__
+      const requestId = window?.__REQUEST_ID__
+      const defaultError =
+        'Render Runtime renderered an error page and there is no error or request id available'
+
+      const runtime = window?.global?.__RUNTIME__ ?? {}
+
+      const {
+        account = 'N/A',
+        workspace = 'N/A',
+        culture: { locale } = { locale: 'N/A' },
+        route: { path } = { path: 'N/A' },
+        loadedDevices = ['N/A'],
+        production = 'N/A',
+      } = runtime
+
+      const errorInfo = {
+        account,
+        workspace,
+        locale,
+        path,
+        device: loadedDevices[0],
+        production,
+      }
 
-    if (!isAdmin()) {
-      return
-    }
```

Comment:
I would move this to the top, right before the:

```js
window.setTimeout(() => {
      this.setState({ enabled: true })
    }, 5000)
```

Reply:
Do you mean something like that?

```ts
    window.setTimeout(() => {
      this.setState({ enabled: true })
    }, 5000)
   
    try {
      const error = window?.__ERROR__
      const requestId = window?.__REQUEST_ID__
      const defaultError =
        'Render Runtime renderered an error page and there is no error or request id available'

      const runtime = window?.global?.__RUNTIME__ ?? {}
      const errorInfo = this.extractErrorInfo(runtime)

      if (!isAdmin() || errorInfo.production === false) {
        return
      }

      getCurrentScope().setExtras(errorInfo)

      if (error) {
        captureException(error)
      } else if (requestId) {
        captureException(requestId)
      } else {
        captureException(defaultError)
      }
    } catch (e) {
      if (!isAdmin()) return

      captureException(e)
    }
```